extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java

```diff
@@ -527,8 +529,12 @@ private static boolean collectDeclaredProvidersForMethodAndMediaTypeAnnotation(S
                 return true;
             }
         }
+        String[] mediaTypes = WILDCARD_MEDIA_TYPE_ARRAY;
+        if (mediaTypeMethodAnnotationInstance.value() != null) {
```

Comment:
Hmmm, so Jandex doesn't give you the default value? That sounds like an issue in Jandex, doesn't it?

Reply:
It doesn't. But that is something we've dealt with in most Quarkus extensions explicitly.

@mkouba @n1hility would it be possible for Jandex to respect the default value set in the annotation source code?

Reply:
You should use `org.jboss.jandex.AnnotationInstance.valueWithDefault(IndexView, String)` if you need the default value. The reason why you need the index is that the default values are stored in the annotation class definition whereas the `AnnotationInstance` only stores the annotation name.

Reply:
Gotcha, thanks

Reply:
In this case I don't think it makes sense to use that though, because `@Produces` is not in any index and we get:

```
java.lang.IllegalArgumentException: Index did not contain annotation definition: javax.ws.rs.Produces

```

So I think what the PR does currently makes the most sense